umich_api/api_utils.py

```diff
@@ -75,13 +82,17 @@ def api_call(self, api_call, client_scope, method="GET", payload=None):
         :returns Requests object with response from API (or an Exception)
         :raises Exception: No Access Token
         """
-        return self.scopes[client_scope]["api_call"](api_call, self.scopes[client_scope]["access_token"], method, payload)
+        this_scope = self.scopes.get(client_scope)
```

Comment:
make sure the `this_scope` is not empty/None before you make the `this_scope.get` call otherwise it will lead to exception

Reply:
Yeah, I could check more None returns and raise an exception. Is this better than having an AttributeError raised on the NoneType? I have to do some research on best practices.